src/toggle-switch/toggle-switch.scss

```diff
@@ -0,0 +1,235 @@
+  background-color: var(--color-switch-track-checked-bg);
+  border-color: var(--color-switch-track-checked-border);
+
+  &:hover {
+    background-color: var(--color-switch-track-checked-hover-bg);
+  }
+
+  &:active {
+    background-color: var(--color-switch-track-checked-active-bg);
+  }
+
+  .ToggleSwitch-knob {
+    background-color: var(--color-switch-knob-checked-bg);
+    border: 0;
+    transform: translateX(calc(100% + 1px));
+  }
+
+  .ToggleSwitch-lineIcon {
+    transform: translateX(0%);
+  }
+
+  .ToggleSwitch-circleIcon {
+    transform: translateX(100%);
+  }
+}
+
+.ToggleSwitch-track[aria-disabled='true'] {
+  cursor: not-allowed;
+  background-color: var(--color-canvas-subtle);
+  border-color: var(--color-border-subtle);
+  transition-property: none;
+
+  &:hover,
+  &:active {
+    .ToggleSwitch-knob {
+      background-color: var(--color-btn-bg);
+    }
+  }
+
+  .ToggleSwitch-knob {
+    border-color: var(--color-border-default);
+    box-shadow: none;
+
+    &:hover,
+    &:active {
+      background-color: var(--color-btn-bg);
+    }
+  }
+
+  .ToggleSwitch-lineIcon {
+    color: var(--color-fg-subtle);
+  }
+
+  .ToggleSwitch-circleIcon {
+    color: var(--color-fg-subtle);
+  }
+
+  .ToggleSwitch-labels {
+    color: var(--color-fg-muted);
+  }
+}
+
+.ToggleSwitch-bg {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+}
+
+.ToggleSwitch-lineIcon {
+  line-height: 0;
+  color: var(--color-accent-fg);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(-100%);
+  flex-shrink: 0;
+  flex-grow: 1;
+  flex-basis: 50%;
+}
+
+.ToggleSwitch-circleIcon {
+  line-height: 0;
+  color: var(--color-fg-default);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(0);
+  flex-shrink: 0;
+  flex-grow: 1;
+  flex-basis: 50%;
+}
+
+.ToggleSwitch-knob {
+  position: absolute;
+  top: -1px;
+  bottom: -1px;
+  z-index: 1;
+  width: 50%;
+  background-color: var(--color-btn-bg);
+  border-color: var(--color-switch-track-border);
+  border-style: $border-style;
+  border-width: $border-width;
+  border-radius: $border-radius;
+  box-shadow: var(--color-shadow-medium), var(--color-btn-inset-shadow);
+  transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(-1px);
+
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+
+.ToggleSwitch-labels {
+  position: relative;
+  margin-right: $spacer-2;
+  margin-left: $spacer-2;
+  font-size: $body-font-size;
+  line-height: $body-line-height;
+  color: var(--color-fg-default);
+}
+
+.ToggleSwitch--small {
+  .ToggleSwitch-labels {
+    font-size: $font-size-small;
+  }
+
+  .ToggleSwitch-track {
+    width: 48px;
+    height: 24px;
+  }
+}
+
+.ToggleSwitch-onLabel {
+  height: 0;
+  visibility: hidden;
+}
+
+.ToggleSwitch-offLabel {
```

Comment:
Should the text alignment of these labels respect if they are on the left or right of the toggle? So if its on the right, I think the text should be `text-align: left`

It won't be obvious, but in an extreme case right now it would look like this:
<img width="141" alt="image" src="https://user-images.githubusercontent.com/18661030/169180296-a9eb4085-ac36-43e1-be4d-4d14924ac3a6.png">

Also, I know you are using utilities now but when the label alignment styles move here to CSS.. should we actually render the label before or after the button instead of using flex to manipulate the visual location? Maybe @ichelsea could weigh in on that since usually we want to avoid visually manipulating the dom tree without actually changing the order.

Reply:
Hmm good questions @langermank. Adding `text-align: left` makes sense. I've added it to `.ToggleSwtich--statusAtEnd`.

As for whether to render the status label first or last in-source, I'll defer to @ichelsea :)

Reply:
For this, we will actually want this On / Off text to be hidden from screen readers. The button itself will give the context that they need, and with the proper roles it should read out as On / Off instead of "selected / unselected". The label of the the toggle itself, the one that says what the toggle controls, (make sure that label is programmatically tied to the toggle!), and the toggle functionality itself are the most important ones here.

Reply:
Ok awesome, thanks @ichelsea! The `<button>` element has a role of "switch" right now, is that enough for screen readers to read out "on" and "off"?

I can add `aria-hidden="true"` to the label.